### DISCUSSION TITLE
Praise

### DISCUSSION THREAD

**Discussion by dyst5422** [bikeshaving/crank#32] [General]
For lack of a better forum to do this, let's gather all deserved praise in this issue.

**Comment by dyst5422** [comment]
This is freaking awesome. I can't even express how amazing this is. It is so so clean and so so intuitive (well aside from the event listener api, but I've been just using onclick etc anyways to get around that).

I love how freeing the API is - literally just javascript that returns JSX. It doesn't care how the JSX is returned, it just does it.

I wish this had come around earlier in the lifetime of react because I think this would have been the way they would have chosen to go with it.

I DO think that the one element that react has over this is when it comes to async rendering regarding pausing/resuming large diffing and prioritization (which I firmly believe should be resolved in the v8 space, not the javascript space).

Bravo! This is the kind of thing that makes me love programming.

**Comment by ryhinchey** [comment]
+1 so excited for this project!

**Comment by plusgut** [comment]
This project is indeed awesome, I really like the approach to rethink and improve on the APIs we are so used to.
I applaud @brainkim for this interesting project. Even though that I don't agree with all the decisions he made, I am still very grateful.

**Comment by benjamingr** [comment]
Yeah this is a really nice idea, I did some early work on async iteration and reviewed some materials. I even did some PoCs of this idea [a while ago](https://gist.github.com/benjamingr/5c55e68efd1d154858823aa7c8ef6827) and [here](https://gist.github.com/benjamingr/81aa5a87cf3bf282c36b2ac0f6f69a5b) If you want to discuss API ideas I'd love to discuss them.

**Comment by mcjazzyfunky** [comment]
Please allow me not only to thank @brainkim but also the whole Crank community for all the input to the project.
It's really a pleasure to take part in discussions within the Crank project and sharing thoughts.
Everybody is so extraordinary friendly and helpful and full of ideas - it's really a joy.
Have a good time and stay healthy.

**Comment by brainkim** [comment]
Thanks for the kind words! The initial feedback was really overwhelming and I’m grateful to all of you for providing kind and insightful thoughts and comments. Also, I promise that @dyst5422 isn’t my alt account 😄I’m going to convert this issue to a discussion to test out the new feature (We got GitHub discussions enabled for crank 🚀) so feel free to continue the conversation or whatever. Or don’t, I also like criticism and getting into arguments about `this` or the `EventTarget` interface too.

**Comment by jazzdev** [comment]
Yeah, Kudos for writing Crank.  l like, simple, light-weight frameworks.
